services/system/AuthAny/src/AuthAny.cpp

```diff
@@ -20,26 +20,29 @@ namespace SystemService
                               std::vector<Claim>         claims)
    {
       if (enable_print)
-         std::printf("auth_check\n");
+         std::printf("checkAuthSys\n");
    }
 
    void AuthAny::canAuthUserSys(AccountNumber user)
    {
       if (enable_print)
-         std::printf("user_check\n");
+         std::printf("canAuthUserSys\n");
    }
 
-   void AuthAny::stagedAccept(uint32_t stagedTxId, AccountNumber actor)
+   bool AuthAny::isAuthSys(AccountNumber sender, std::vector<AccountNumber> authorizers)
    {
-      check(getSender() == StagedTxService::service, "can only be called by staged-tx");
+      if (enable_print)
+         std::printf("isAuthSys\n");
 
-      auto [execute, allowedActions] = to<StagedTxService>().get_exec_info(stagedTxId);
-      recurse().to<Transact>().runAs(std::move(execute), allowedActions);
+      return true;
    }
 
-   void AuthAny::stagedReject(uint32_t stagedTxId, AccountNumber actor)
+   bool AuthAny::isRejectSys(AccountNumber sender, std::vector<AccountNumber> rejecters)
    {
-      check(false, "not supported");
+      if (enable_print)
+         std::printf("isRejectSys\n");
+
+      return true;
```

Comment:
Is this the right behavior? My understanding of the expected sematics of reject is "return true iff no set of accepts that does not conflict with the rejects will be sufficient to authorize"

Reply:
Hmm, I don't think accepts ever need to be considered during reject. We know the accept criteria wasn't met, otherwise it would already be accepted and this wouldn't be called. So we just need to check if a reject threshold was met. 

In `auth-any` specifically, this would only be called via indirection (when an auth-any account is on an msig for example), so the intention of returning true here is to imply that `authAny` always agrees with what everyone else wants.

Reply:
Here's an example:

A transaction contains two actions:
action1 is authorized by msig 2/3 alice, bob, carol
action2 is authorized by auth-any

carol rejects the transaction

Reply:
My point about accept was what I expect the relationship between the accept and reject thresholds to be.

reject(S, A) = ¬accept(S, A')

Reply:
>  what I expect the relationship between the accept and reject thresholds to be

The thresholds can be whatever... and can be user-defined, for example an msig may require super-majority to accept, simple-majority to reject

Reply:
I don't see reject as an independent operation. It is the opposite of accept, that is, it indicates that the rejector will not accept the transaction. Deleting the transaction is just a convenience that should happen when the authorization becomes unsatisfiable under that assumption.

Reply:
Being able to satisfy both the accept and reject thresholds simultaneously is just confusing. It's ambiguous and the actual result depends on the order in which they are submitted, which creates a race.

Reply:
> Didn't address this yet because I still don't understand.

Let's say that in addition to carol rejecting, alice and bob accept. There are two things that I find weird:
- Whether it is accepted or rejected depends on whether alice and bob are quick enough to beat carol.
- Adding an action unrelated to carol caused carol's reject to reject the entire transaction.

Reply:
> There are two things that I find weird:

Okay, I agree with both of these things. I find them unimportant because this is a problem unique to people who are proposing transactions that use an auth-any account, which shouldn't really be a thing.

Should I just make auth-any always return false from reject?

Reply:
> Should I just make auth-any always return false from reject?

I think so.

Reply:
Done